docs/plugins/analysis-smartcn.asciidoc

```diff
@@ -17,7 +17,433 @@ include::install_remove.asciidoc[]
 
-NOTE: The `smartcn_word` token filter and `smartcn_sentence` have been deprecated.
+==== Reimplementing and extending the analyzers
+
+The `smartcn` analyzer could be reimplemented as a `custom` analyzer that can
+then be extended and configured as follows:
+
+[source,js]
+----------------------------------------------------
+PUT smartcn_example
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "rebuilt_smartcn": {
+          "tokenizer":  "smartcn_tokenizer",
+          "filter": [
+            "porter_stem",
+            "smartcn_stop"
+          ]
+        }
+      }
+    }
+  }
+}
+----------------------------------------------------
+// CONSOLE
+// TEST[s/\n$/\nstartyaml\n  - compare_analyzers: {index: smartcn_example, first: smartcn, second: rebuilt_smartcn}\nendyaml\n/]
+
+[[analysis-smartcn_stop]]
+==== `smartcn_stop` token filter
+
+The `smartcn_stop` token filter filters out stopwords defined by `smartcn`
+analyzer (`_smartcn_`), and any other custom stopwords specified by the user.
+This filter only supports the predefined `_smartcn_` stopwords list.
+If you want to use a different predefined list, then use the
+{ref}/analysis-stop-tokenfilter.html[`stop` token filter] instead.
+
+[source,js]
+--------------------------------------------------
+PUT smartcn_example
+{
+  "settings": {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "smartcn_with_stop": {
+            "tokenizer": "smartcn_tokenizer",
+            "filter": [
+              "porter_stem",
+              "my_smartcn_stop"
+            ]
+          }
+        },
+        "filter": {
+          "my_smartcn_stop": {
+            "type": "smartcn_stop",
+            "stopwords": [
+              "_smartcn_",
+              "stack"
+            ]
+          }
+        }
+      }
+    }
+  }
+}
+
+GET smartcn_example/_analyze
+{
+  "analyzer": "smartcn_with_stop",
+  "text": "哈喽，我们是 Elastic   我们是 Elastic Stack（Elasticsearch、Kibana、Beats 和 Logstash）的开发公司。从股票行情到 Twitter 消息流，从 Apache 日志到 WordPress 博文，我们可以帮助人们体验搜索的强大力量，帮助他们以截然不同的方式探索和分析数据"
+}
+--------------------------------------------------
+// CONSOLE
+
+The above request returns:
+
+[source,js]
+--------------------------------------------------
+{
+    "tokens": [
+        {
+            "token": "哈",
+            "start_offset": 0,
+            "end_offset": 1,
+            "type": "word",
+            "position": 0
+        },
+        {
+            "token": "喽",
+            "start_offset": 1,
+            "end_offset": 2,
+            "type": "word",
+            "position": 1
+        },
+        {
+            "token": "我们",
+            "start_offset": 3,
+            "end_offset": 5,
+            "type": "word",
+            "position": 3
+        },
+        {
+            "token": "是",
+            "start_offset": 5,
+            "end_offset": 6,
+            "type": "word",
+            "position": 4
+        },
+        {
+            "token": "elast",
```

Comment:
this result seems not good, `elastic`->`elast`

Reply:
@medcl Thanks. this is expected and it happens because of the porter stemmer, which is a part of the [SmartChineseAnalyzer](https://github.com/apache/lucene-solr/blob/master/lucene/analysis/smartcn/src/java/org/apache/lucene/analysis/cn/smart/SmartChineseAnalyzer.java#L139).  As this example meant to demonstrate all parts of the SmartChineseAnalyzer, I included the porter stemmer as well